crypto/compactcert/builder.go

```diff
@@ -41,88 +42,102 @@ type sigslot struct {
 // Builder keeps track of signatures on a message and eventually produces
 // a compact certificate for that message.
 type Builder struct {
-	Params
-	sigs          []sigslot // Indexed by pos in participants
-	sigsHasValidL bool      // The L values in sigs are consistent with weights
-	signedWeight  uint64    // Total weight of signatures so far
-	participants  []basics.Participant
-	parttree      *merklearray.Tree
-
-	// Cached cert, if Build() was called and no subsequent
-	// Add() calls were made.
-	cert *Cert
+	data           StateProofMessageHash
+	round          uint64
+	sigs           []sigslot // Indexed by pos in participants
+	signedWeight   uint64    // Total weight of signatures so far
+	participants   []basics.Participant
+	parttree       *merklearray.Tree
+	lnProvenWeight uint64
+	provenWeight   uint64
+	strengthTarget uint64
 }
 
-// MkBuilder constructs an empty builder (with no signatures).  The message
-// to be signed, as well as other security parameters, are specified in
-// param.  The participants that will sign the message are in part and
-// parttree.
-func MkBuilder(param Params, part []basics.Participant, parttree *merklearray.Tree) (*Builder, error) {
+// Errors for the CompactCert builder
+var (
+	ErrPositionOutOfBound     = errors.New("requested position is out of bound")
+	ErrPositionWithZeroWeight = errors.New("position has zero weight")
+	ErrInternalCoinIndexError = errors.New("error while calculate coin to index")
```

Comment:
I think this error message should indicate something more serious, since this error should never happen. It is an unexpected behavior, due to a bug outside this function, and not a calculation error inside the reporting function. 
```suggestion
	ErrInternalCoinIndexError = errors.New("error while calculating coin to index")
```